docs/adrs/0002-uds-core-functional-layers.md

```diff
@@ -0,0 +1,73 @@
+# ADR: UDS Core Functional Layers
+
+Date: 2024-07-17
+
+## Status
+
+Accepted
+
+## Context
+
+Historically, UDS Core intentionally only published a single official Zarf Package to ensure a standard baseline across all UDS environments. We learned form the complex buffet menu-style of Big Bang that it is better to form opinions and test them than to allow unlimited configuration possibilities. We have also found the monorepo structure of UDS Core to be be much simpler for developing UDS Core. However, through user feedback, we have seen cases where more optionality is warranted, but having to modify UDS CLI to accommodate disabling portions of UDS Core is not ideal. We also have a need for an edge-compatible version of UDS Core that is more lightweight and does not include all the components of the full UDS Core package.
+
+Today, we publish both the official UDS Core package and a UDS Slim Dev package, in addition to dev and demo UDS Bundles. The UDS Slim Dev package is a stripped-down version of UDS Core that includes limited services meant for testing UDS Packages. We have also learned that some teams outside of UDS need similar functionality, but slightly different parts of UDS Core, such as keycloak + authsvc.
+
+## Decision
+
+We will replace the existing 2-package (one dev and one prod) structure with a layered set of packages within the existing monorepo. This will allow users to choose the components they need for their specific use case, while maintaining a single source of truth for all UDS Core components. We will not break these out by applications as Big Bang did, but instead by layers of functionality for related/integrated components. This allows more flexibility for which functionality is provided by UDS Core and which is provided by other external packages.
+
+The functional layers will be:
+
+1. UDS Core Base (Istio + Pepr w/UDS Operator & UDS Policies)
+2. UDS Core Identity and Authorization (Keycloak + AuthService)
```

Comment:
Do we have timing metrics on auth service (not opposed to its inclusion though wanted to bring up the concerns @zachariahmiller had had on it being on by default/inseparable)

Reply:
I think we can work to optimize it (as far as I know keycloak is actually the slow boot), but the reality is functional-based authsvc on its own it not super useful and not something we want to support (realize SWF doesnt need authsvc rn)

Reply:
yeah KC is def the slowdown today (~60s of the 5 mins to deploy uds-core) - if authservice is only a few seconds and a few milli-CPUs I can see it being useful even for us - we have talked about making podinfo (in uds-common) a better example and that would need authservice.

Reply:
We're certainly on the edge of having to go off free runners for CI on a couple of packages, so theoretically even though authservice only takes ~10 seconds to come up it could push us over that threshold, but if switching to the large runners in that event is an acceptable cost then its fine.